File: projects/demos/engineering.purdue.edu/scriptingwithobjects/swocode/chap3/SigHandler.py
#!/usr/bin/python

### SigHandler.py

import signal                                                       #(A)
import os                                                           #(B)

def ctrl_c_handler( signum, frame ):                                #(C)
    print "Ctrl C pressed"                                          #(D)
    os.kill( os.getpid(), signal.SIGKILL )                          #(E)

signal.signal( signal.SIGINT, ctrl_c_handler )                      #(F)

while (True): pass                                                  #(G)

File: projects/demos/engineering.purdue.edu/scriptingwithobjects/swocode/chap2/SigHandler.pl
#!/usr/bin/perl -w
use strict;

# SigHandler.pl

sub ctrl_c_handler {                                                  #(A)
    print "Ctrl C pressed\n";                                         #(B)
    kill 9, $$;                                                       #(C)
}

$SIG{ "INT" } = 'ctrl_c_handler';                                     #(D)

#$SIG{ "INT" } = \&ctrl_c_handler;                                    #(E)

while (1) {}                                                          #(F)
